notebook/static/notebook/js/cell.js

```diff
@@ -515,14 +516,28 @@ define([
     };
 
     /**
-     * is the cell deletable? only false (undeletable) if
-     * metadata.deletable is explicitly false -- everything else
+     * is the cell edtitable? only false (readonly) if
+     * metadata.editable is explicitly false -- everything else
      * counts as true
      *
+     * @method is_editable
+     **/
+    Cell.prototype.is_editable = function () {
+        if (this.metadata.editable === false) {
+            return false;
+        }
+        return true;
+    };
+
+    /**
+     * is the cell deletable? only false (undeletable) if
+     * metadata.deletable is explicitly false or if the cell is not
+     * editable -- everything else counts as true
+     *
      * @method is_deletable
      **/
     Cell.prototype.is_deletable = function () {
-        if (this.metadata.deletable === false) {
+        if (this.metadata.deletable === false || !this.is_editable()) {
```

Comment:
Yes, I think it should update the old metadata. As @minrk said:

> Yup, any change like that should include automatic upgrading of the old format on load.

So even before it gets to the point of checking if the cell is deleteable, ideally the old metadata format should be updated.